app/src/main/java/com/example/tasky/agenda/presentation/viewmodel/AgendaViewModel.kt

```diff
@@ -26,33 +28,120 @@ class AgendaViewModel @Inject constructor(
     private val agendaRepository: AgendaRepository,
 ) : ViewModel() {
 
+    private val selectedDate: MutableStateFlow<LocalDate> = MutableStateFlow(LocalDate.now())
+
     var state by mutableStateOf(AgendaState())
         private set
 
     init {
         state = state.copy(
-            days = dateGenerator.getWeek(),
             userInitials = GetInitialsUseCase(userCache.getUser()?.fullName),
-            selectedMonth = dateGenerator.getMonth()
         )
-        val now = LocalDate.now()
         viewModelScope.launch {
-            agendaRepository.getAgenda(now).collectLatest {
+            selectedDate.collect {
```

Comment:
I would use collectLatest, since you want the last emission of the flow

Also something extra, when getting a flow I like to add a "distinctUntilChanged()" as to avoid useless collections

So in this case:
```
selectedDate.distinctUntilChanged().collectLatest { ... }
```

Reply:
![image](https://github.com/MelvinKucuk/tasky/assets/42274125/f07fd9fc-7452-434c-96b8-36987dc91085)
I get this warning